core/field_colour_slider.js

```diff
@@ -45,343 +35,307 @@ goog.require('goog.ui.Slider');
+   * BEWARE: This is not a stable API, so it is being marked as private. It may change.
+   * @private
+   */
+  static activateEyedropper_ = null;
+
+  constructor(colour, opt_validator) {
+    super(colour, opt_validator);
+
+    /**
+     * Path to the eyedropper svg icon.
+     */
+    this.EYEDROPPER_PATH = 'eyedropper.svg';
+    this.SERIALIZABLE = true;
+    this.EDITABLE = true;
   }
-  Blockly.FieldColourSlider.superClass_.init.call(this, block);
-  this.setValue(this.getValue());
-};
 
-/**
- * Return the current colour.
- * @return {string} Current colour in '#rrggbb' format.
- */
-Blockly.FieldColourSlider.prototype.getValue = function() {
-  return this.colour_;
-};
-
-/**
- * Set the colour.
- * @param {string} colour The new colour in '#rrggbb' format.
- */
-Blockly.FieldColourSlider.prototype.setValue = function(colour) {
-  if (this.sourceBlock_ && Blockly.Events.isEnabled() &&
-      this.colour_ != colour) {
-    Blockly.Events.fire(new Blockly.Events.BlockChange(
-        this.sourceBlock_, 'field', this.name, this.colour_, colour));
+  /**
+   * Construct a FieldColourSlider from a JSON arg object.
+   * @param {!Object} options A JSON object with options (colour).
+   * @returns {!Blockly.FieldColourSlider} The new field instance.
+   * @package
+   * @nocollapse
+   */
+  static fromJson(options) {
+    return new FieldColourSlider(options['colour']);
   }
-  this.colour_ = colour;
-  if (this.sourceBlock_) {
-    // Set the colours to this value.
-    // The renderer expects to be able to use the secondary colour as the fill for a shadow.
-    this.sourceBlock_.setColour(colour, colour, this.sourceBlock_.getColourTertiary(),
-        this.sourceBlock_.getColourQuaternary());
-  }
-  this.updateSliderHandles_();
-  this.updateDom_();
-};
 
-/**
- * Create the hue, saturation or value CSS gradient for the slide backgrounds.
- * @param {string} channel – Either "hue", "saturation" or "value".
- * @return {string} Array colour hex colour stops for the given channel
- * @private
- */
-Blockly.FieldColourSlider.prototype.createColourStops_ = function(channel) {
-  var stops = [];
-  for(var n = 0; n <= 360; n += 20) {
-    switch (channel) {
-      case 'hue':
-        stops.push(goog.color.hsvToHex(n, this.saturation_, this.brightness_));
-        break;
-      case 'saturation':
-        stops.push(goog.color.hsvToHex(this.hue_, n / 360, this.brightness_));
-        break;
-      case 'brightness':
-        stops.push(goog.color.hsvToHex(this.hue_, this.saturation_, 255 * n / 360));
-        break;
-      default:
-        throw new Error("Unknown channel for colour sliders: " + channel);
+  doValueUpdate_(newValue) {
+    super.doValueUpdate_(newValue);
+    if (this.sourceBlock_) {
+      // Set the colours to this value.
+      // The renderer expects to be able to use the secondary colour as the fill for a shadow.
+      this.sourceBlock_.setColour(newValue);
```

Comment:
If you remove this does it continue to work? I think the superclass sets the colour if you're running zelos.

https://github.com/google/blockly/blob/e2df0fc2885cb994886c0af863035f50bb849948/core/field_colour.ts#L214

Reply:
Yep, done!